ipa-core/src/query/runner/hybrid.rs

```diff
@@ -61,35 +58,24 @@ where
             ));
         }
 
-        let (_decrypted_reports, tags): (Vec<HybridReport<BA8, BA3>>, Vec<UniqueTag>) =
-            LengthDelimitedStream::<EncryptedHybridReport, _>::new(input_stream)
-                .map_err(Into::<Error>::into)
-                .map_ok(|enc_reports| {
-                    iter(enc_reports.into_iter().map({
-                        |enc_report| {
-                            let dec_report = enc_report
-                                .decrypt::<R, BA8, BA3, BA20>(key_registry.as_ref())
-                                .map_err(Into::<Error>::into);
-                            let unique_tag = UniqueTag::from_unique_bytes(&enc_report);
-                            dec_report.map(|dec_report1| (dec_report1, unique_tag))
-                        }
-                    }))
-                })
-                .try_flatten()
-                .take(sz)
-                .try_fold(
-                    (Vec::with_capacity(sz), Vec::with_capacity(sz)),
-                    |mut acc, result| async move {
-                        acc.0.push(result.0);
-                        acc.1.push(result.1);
-                        Ok(acc)
-                    },
-                )
-                .await?;
-
-        let resharded_tags = reshard_iter(
+        let stream = LengthDelimitedStream::<EncryptedHybridReport, _>::new(input_stream)
+            .map_err(Into::<Error>::into)
+            .map_ok(|enc_reports| {
```

Comment:
one of the big challenges I've had thus far is how to reason about `enc_reports` and how it could interact with reshard. but collection all of these into a stream solves that and makes it much easier to deal with.

I wonder if we should consider hiding some of that complexity "under the hood" in `LengthDelimitedStream` and have it operate like a "normal" stream that where `.next()` gives an individual item, and not a Vec<item>. not in scope for this PR, but curious of others opinions here.